src/app/(main)/dashboard/components/voyage-dashboard/getDashboardData.ts

```diff
@@ -43,43 +96,69 @@ export const getDashboardData = async (
   });
 
   if (errorResponse) {
-    return {
-      currentSprintNumber: null,
-      sprintsData: [],
-      user: null,
-      meetingsData: [],
-      voyageNumber: null,
-      voyageData: {} as Voyage,
-      errorMessage: errorResponse,
-    };
+    return errorResponse;
   }
 
   if (data) {
     const [res, error] = data;
 
     if (error) {
-      throw new Error(`Error: ${error.message}`);
+      return `Error: ${error.message}`;
     }
 
     sprintsData = res!.sprints;
     voyageNumber = Number(res!.number);
     voyageData = res!;
   }
 
+  return { sprintsData, voyageNumber, voyageData };
+};
+
+export const getDashboardData = async (
+  user: User | null,
+  error: AppError | null,
+  teamId: number,
+): Promise<GetDashboardDataResponse> => {
+  const sprintsResult = await getSprintsData(user, error, teamId);
+  const featuresResult = await fetchData<FeaturesList[], { teamId: number }>(
+    fetchFeatures,
+    user,
+    error,
+    teamId,
+    { teamId },
+  );
+  const projectIdeasResult = await fetchData<
+    IdeationData[],
+    { teamId: number }
+  >(fetchProjectIdeas, user, error, teamId, { teamId });
+  const techStackResult = await fetchData<TechStackData[], { teamId: number }>(
+    fetchTechStack,
+    user,
+    error,
+    teamId,
+    { teamId },
+  );
+  const resourcesResult = await fetchData<ResourceData[], { teamId: number }>(
+    fetchResources,
+    user,
+    error,
+    teamId,
+    { teamId },
+  );
+
+  if (typeof sprintsResult === "string") throw new Error(sprintsResult);
+  if (featuresResult.error) throw new Error(featuresResult.error);
+  if (projectIdeasResult.error) throw new Error(projectIdeasResult.error);
+  if (techStackResult.error) throw new Error(techStackResult.error);
+  if (resourcesResult.error) throw new Error(resourcesResult.error);
```

Comment:
Jane made some updates to error handling so this branch will need to be updated to fit that, but in addition to that, don't throw any errors like this.